linux_os/guide/system/logging/rsyslog_filecreatemode/bash/shared.sh

```diff
@@ -0,0 +1,5 @@
+# platform = multi_platform_all
+
+sed -i '/^\s*$FileCreateMode/d' /etc/rsyslog.conf /etc/rsyslog.d/*
```

Comment:
Thinking about this remediation, wouldn't it be better to first check if the value is correct and only change if necessary? If the admin decided to configure it correctly, but in a different file, by any reason, the remediation would unnecessarily change the file. This might be undesired in some cases.

Reply:
let me know if you have thoughts about it or if you did it differently for the ansible

Reply:
In Ansible, the logic would be something like:
- Find all rsyslog files containing $FileCreateMode
  - There is a rule using a similar approach:
  - https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/logging/rsyslog_sending_messages/rsyslog_remote_tls/rule.yml
- Concatenate these files in a list.
- If the list contains more than 1 file, remove the parameter in all files and create a new file specific for the parameter, in order to avoid conflicts.
- If the list contains only 1 file, ensure the value is correct on that file. Just update the value, if necessary.
- If the list is empty, create a new file with the parameter.

I believe a similar logic, maybe simpler, could be used in bash to make it more idempotent.
Basically, it would be necessary to only implement these tests in the script.

Reply:
sorry for the delay, I've just rebased the branch and improved the remediation.
Let me know if that looks good.